app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarMenuController.kt

```diff
@@ -295,6 +298,16 @@ class DefaultBrowserToolbarMenuController(
         }
     }
 
+    private fun getProperUrl(currentSession: Session?): String? {
+        return currentSession?.id?.let {
+            if (browserStore.state.findTab(it)?.readerState?.active == true) {
+                browserStore.state.findTab(it)?.readerState?.activeUrl
```

Comment:
nit: this isn't super expensive, but I would still only call `findTab` once, by introducing a local `val tab = store.state.findTab(it)`